packages/virtualized-lists/Lists/VirtualizedList.js

```diff
@@ -1643,6 +1718,11 @@ class VirtualizedList extends StateSafePureComponent<Props, State> {
       visibleLength,
       zoomScale,
     };
+    if (this.state.maintainVisibleContentPositionAdjustment != null) {
+      this.setState({
+        maintainVisibleContentPositionAdjustment: null,
```

Comment:
This may not be correct in the following sequence:
1. We change data once
2. We change data again
3. We get a scroll update from #&#8203;1

Consider a counter instead like `pendingScrollUpdateCount: number`.

Reply:
Good catch on this, I was able to reproduce this case by adding pages multiple times with a small delay

```js
const onPress = () => {
  addPages();
  setTimeout(() => {
    addPages();
  }, 1);
}